Backend/database/firestore/database_manager.py

```diff
@@ -0,0 +1,19 @@
+import firebase_admin
+from firebase_admin import credentials
+from firebase_admin import firestore_async
+
+cred = credentials.Certificate("./database/firestore/credentials.json")
+
+# Initialize the Firebase Admin SDK
+firebase_admin.initialize_app(cred)
```

Comment:
Maybe we could move this to DatabaseManager?

Reply:
pull it inside generate few problems:
1) any request attempt try to generate new connection to database and then throw error
2) if you wrap it into `try...except` it generate redundant requests to database 

it is why I decide to move it before

Reply:
Oh, I see
Another idea I have to repair it is to make `DatabaseManager` singleton with late initialization of `self.db`. If you want you could try this code, but I'm okay with keeping it as it is. 

```python
import firebase_admin
from firebase_admin import credentials, firestore_async


class DatabaseManager:
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super(DatabaseManager, cls).__new__(cls)
            cls._instance.db = None
        return cls._instance

    def init_db(self):
        if self.db is None:
            # Initialize Firebase here
            cred = credentials.Certificate("./database/firestore/credentials.json")
            firebase_admin.initialize_app(cred)

            # Initialize Firestore client
            self.db = firestore_async.client()
            if self.db is None:
                raise RuntimeError("Error creating Firestore client")

    def get_firestore_client(self):
        if self.db is None:
            self.init_db()
        return self.db

# Usage
db_manager = DatabaseManager()
firestore_client = db_manager.get_firestore_client()
```